### PR TITLE
test(relayer): adds initial testing base

### PR DESCRIPTION
**Pull request by fcanela**
This PR introduces a foundation consisting of basic setup and an initial test for relayer.

- Adds Jest, with types and coverage output
- Implements basic tests for ShutdownManager to validate the stack

---

## PR-Codex overview
This PR introduces testing capabilities using `jest`, adds a new `ShutdownManager` class, and updates configuration files to support TypeScript and ESLint. It also enhances coverage reporting and updates various package dependencies.

### Detailed summary
- Added `ShutdownManager` class in `shutdownManager.ts`.
- Implemented tests for `ShutdownManager` in `shutdownManager.test.ts`.
- Updated `.eslintrc.json` to include `jest` and `ts-jest`.
- Added `jest` and `@types/jest` to `devDependencies`.
- Configured `jest` in `jest.config.ts` for TypeScript.
- Modified `package.json` to include a test script for coverage.
- Updated various dependencies in `yarn.lock`.

> The following files were skipped due to too many changes: `yarn.lock`

> ✨ Ask PR-Codex anything about this PR by commenting with `/codex {your question}`

## Release Notes

- **New Features**
  - Introduced a new `ShutdownManager` class to manage system shutdown states.
  - Added a Jest configuration file for improved testing capabilities, including coverage reporting.
  - New test script added to `package.json` for running tests with coverage.
  - Updated ESLint configuration to recognize Jest-specific globals.

- **Bug Fixes**
  - Simplified shutdown management logic by externalizing it to the `ShutdownManager` class.

- **Tests**
  - Added unit tests for the `ShutdownManager` class, covering instance creation and shutdown status retrieval.
  - Included tests for the `getIsShuttingDown` method and a placeholder for the `triggerShutdown` method.

### CODE REVIEW

File: relayer-cli/src/utils/shutdownManager.test.ts [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+
+    it("should return true when isShuttingDown is false", () => {
+      const instance = new ShutdownManager(false);
+      expect(instance.getIsShuttingDown()).toBe(false);
+    });
+  });
+
+  describe("triggerShutdown", () => {
+    it.todo("should set isShuttingDown to true");
+  });
```
[the remark this answers is not in the document]
Comment (fcanela) [reply relayer-cli/src/utils/shutdownManager.test.ts:35]: This was code was created during an internal testing exercise. The effort finishes there. A teammate or myself will take on after this base testing implementation with examples.

File: relayer-cli/src/utils/shutdownManager.test.ts [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+describe("ShutdownManager", () => {
+  describe("constructor", () => {
+    it("should create a new instance", () => {
+      const instance = new ShutdownManager();
+      expect(instance).toBeInstanceOf(ShutdownManager);
+    });
+
+    it("should set isShuttingDown to the provided value", () => {
+      const instance = new ShutdownManager(true);
+      expect(instance["isShuttingDown"]).toBe(true);
```
[the remark this answers is not in the document]
Comment (fcanela) [reply relayer-cli/src/utils/shutdownManager.test.ts:12]: This was provided as an example of how to access private methods/fields on testing, so it is intended

File: relayer-cli/src/utils/shutdownManager.test.ts [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+    it("should return true when isShuttingDown is false", () => {
+      const instance = new ShutdownManager(false);
+      expect(instance.getIsShuttingDown()).toBe(false);
+    });
+  });
+
+  describe("triggerShutdown", () => {
+    it.todo("should set isShuttingDown to true");
+  });
+});
```
[the remark this answers is not in the document]
Comment (fcanela) [reply relayer-cli/src/utils/shutdownManager.test.ts:36]: This is a good one. I need to improve the `eslint` configuration to recognize `jest` globals.

File: relayer-cli/package.json [hunk 4]
```diff
@@ -23,6 +24,10 @@
     "web3-batched-send": "^1.0.3"
   },
   "devDependencies": {
-    "ts-node": "^10.9.2"
+    "@types/jest": "^29.5.14",
+    "jest": "^29.7.0",
+    "ts-jest": "^29.2.5",
+    "ts-node": "^10.9.2",
+    "yarn": "^1.22.22"
```
Comment (mani99brar) [inline relayer-cli/package.json:31]: Is `yarn` needed to be defined in devDependencies? Also the version differs from the global one, `4.2.2`

Comment (fcanela) [reply relayer-cli/package.json:31]: @mani99brar somehow my local environment decided to add it. It was not intentional. I'm removing it.

File: relayer-cli/src/utils/shutdownManager.test.ts (hunk 3 above)
[the remark this answers is not in the document]
Comment (fcanela) [reply relayer-cli/src/utils/shutdownManager.test.ts:36]: Done in the most recent commit

File: relayer-cli/package.json (hunk 4 above)
Comment (fcanela) [reply relayer-cli/package.json]: Removed in the latest commit

**Review by mani99brar — APPROVED** [review]

**Review by mani99brar — APPROVED** [review]

**Pull request merged**